src/rust/src/x509/extensions.rs

```diff
@@ -416,6 +416,133 @@ fn encode_scts(ext: &pyo3::Bound<'_, pyo3::PyAny>) -> CryptographyResult<Vec<u8>
     Ok(asn1::write_single(&result.as_slice())?)
 }
 
+fn encode_naming_authority<'a>(
+    py: pyo3::Python<'_>,
+    ka_str: &'a cryptography_keepalive::KeepAlive<pyo3::pybacked::PyBackedStr>,
+    py_naming_authority: &pyo3::Bound<'a, pyo3::PyAny>,
+) -> Result<extensions::NamingAuthority<'a>, CryptographyError> {
+    let py_oid = py_naming_authority.getattr(pyo3::intern!(py, "id"))?;
+    let id = if py_oid.is_truthy()? {
```

Comment:
Prefer `is_none()`, it better expresses what we want, and is also a smidge faster.

Reply:
This is a really great hint, thank you! Indeed using it makes sense in all occurences, and by doing that I even uncovered an error I slipped into the codebase in #11881 - the `professionOIDs` field is optional, but I missed that. Will amend this in a separate PR.

Reply:
@alex what I forgot to ask: what do you prefer more?
```rs
let val = if !py_val.is_none() {
    Some(...)
} else {
    None
};
```
or
```rs
let val = if py_val.is_none() {
    None
} else {
    Some(...)
};
```

Reply:
Whichever you prefer, I'm sure we're not consistent